### ISSUE TITLE
`-XBlockArguments`: Avoid Lambda incorrect

### ISSUE THREAD

**Issue by MangoIV** [ndmitchell/hlint#1509]
- avoid lambda with `BlockArguments` doesn't insert parens resulting in an error when applying the hint
- if BlockArguments are on, it should be checkd if the lambda is in parens and if that's not the case, automatically surround the eta reduced expressions with parens.

**Comment by zliu41** [comment]
Some examples would be helpful!

**Comment by MangoIV** [comment]
```haskell
{-# LANGUAGE BlockArguments #-}
module BlockArgumentsHlint where

g :: Int -> a -> Bool
g _ _ = True

r :: (Int -> Bool) -> (Int -> Bool)
r = id

f :: Int -> Bool
f = r \x -> g 3 x
```

**Comment by MangoIV** [comment]
so the correct refactoring would be 
```haskell 
{-# LANGUAGE BlockArguments #-}
module BlockArgumentsHlint where

g :: Int -> a -> Bool
g _ _ = True

r :: (Int -> Bool) -> (Int -> Bool)
r = id

f :: Int -> Bool
f = r (g 3)
```

**Comment by zliu41** [comment]
Thanks! This seems more involved than a good first issue, and may or may not have to do with apply-refact. Will have a closer look later.

**Comment by andreasabel** [comment]
Here is an instance I encountered while working in vscode with the hls:
```haskell
localScope x v = local \ env -> Map.insert x v env
```
The hint is
> Avoid lambda
> Found:
>   \ env -> Map.insert x v env
> Why not:
>   Map.insert x v

If I apply this hint, I am getting the incorrect
```haskell
localScope x v = local Map.insert x v
```
A `$` is needed after `local` or (less preferable) parentheses.

**Comment by zliu41** [comment]
Fix: https://github.com/ndmitchell/hlint/pull/1634

**Comment by MangoIV** [comment]
Thank you for fixing this! ❤️

**Comment by zliu41** [comment]
@MangoIV Thanks for the bug report. Sorry for the delay, it totally slipped my mind.
